### PR TITLE
Serve static client

### PR DESCRIPTION
**Pull request by jernestmyers**
These minor changes do the following:

1. ~add a version of the client's assets in a directory called `./static`~
2. ~configure the `tilemaker` server's `static_directory` setting to serve the client from `./static/` by default~
3. add a catch-all route that will serve static content if `static_directory` is defined

### CODE REVIEW

File: static/assets/index-CBJzntzC.js
Comment (JBorrow) [inline static/assets/index-CBJzntzC.js:1]: Do not package the client here. That packaging should happen on-demand when we package the stuff for upload or for containerisation.

File: static/assets/index-Da165-ik.css
Comment (JBorrow) [inline static/assets/index-Da165-ik.css:1]: Same comment as above

File: static/index.html
Comment (JBorrow) [inline static/index.html:1]: Same comment as above

File: tilemaker/server/app.py [hunk 1]
```diff
@@ -36,15 +36,6 @@
 # Make a simple static dude
 if settings.static_directory is not None:
     app.mount("/static", StaticFiles(directory=settings.static_directory), name="spa")
-    
-    templates = Jinja2Templates(directory="templates")
```
Comment (JBorrow) [inline tilemaker/server/app.py:40]: So I guess you've removed jinja... How do you set the information about the tile server in the client?

File: tilemaker/server/app.py [hunk 2]
```diff
@@ -165,4 +156,11 @@ def histogram_data(
...
-    return response
\ No newline at end of file
+    return response
+
+if settings.static_directory is not None:
+    # catch-all route for static content
+    @app.get("/{full_path:path}")
+    async def serve_spa():
+        index_file_path = os.path.join(settings.static_directory, "index.html")
+        return FileResponse(index_file_path)
```
Comment (JBorrow) [inline tilemaker/server/app.py:166]: Why are you doing this check twice?

File: tilemaker/server/app.py (hunk 1 above)
Comment (jernestmyers) [reply tilemaker/server/app.py:40]: This commit in the client adjusts the vite configs to do so. https://github.com/simonsobs/tileviewer/commit/43ac8d59081c10305bb975a880d6d72aefff4352

File: tilemaker/server/app.py (hunk 2 above)
Comment (jernestmyers) [reply tilemaker/server/app.py:166]: To my understanding, I needed it to be after all other routes in order to serve as a catch-all. Thus, it couldn't share the if-block with the mount logic that, to my understanding, should be at the top of the file. To be fair, I didn't test out alternate placements for mounting the static content though.

**Comment by jernestmyers** [comment]
Update: I modified the `Dockerfile` to build the client assets and copy them into a `/static` directory for the image build.

**Comment by jernestmyers** [comment]
> Is this the version that you have running?

Yes, the Dockerfile in this branch is what's used to build the image being used in [/beta/newsmaps](https://nersc.simonsobs.org/beta/newmaps/). In other words, all the changes from this branch are reflected in the image being used in the now-poorly-named `mapviewerclient` workload on Rancher.

**Pull request merged**